src/diffusers/pipelines/pag/pag_utils.py

```diff
@@ -221,239 +143,82 @@ def _prepare_perturbed_attention_guidance(self, cond, uncond, do_classifier_free
             cond = torch.cat([uncond, cond], dim=0)
         return cond
 
-    def set_pag_applied_layers(self, pag_applied_layers):
-        r"""
-        set the the self-attention layers to apply PAG. Raise ValueError if the input is invalid.
-        """
-
-        if not isinstance(pag_applied_layers, list):
-            pag_applied_layers = [pag_applied_layers]
-
-        for pag_layer in pag_applied_layers:
-            self._check_input_pag_applied_layer(pag_layer)
-
-        self.pag_applied_layers = pag_applied_layers
-
-    @property
-    def pag_scale(self):
-        """
-        Get the scale factor for the perturbed attention guidance.
-        """
-        return self._pag_scale
-
-    @property
-    def pag_adaptive_scale(self):
-        """
-        Get the adaptive scale factor for the perturbed attention guidance.
-        """
-        return self._pag_adaptive_scale
-
-    @property
-    def do_pag_adaptive_scaling(self):
-        """
-        Check if the adaptive scaling is enabled for the perturbed attention guidance.
-        """
-        return self._pag_adaptive_scale > 0 and self._pag_scale > 0 and len(self.pag_applied_layers) > 0
-
-    @property
-    def do_perturbed_attention_guidance(self):
-        """
-        Check if the perturbed attention guidance is enabled.
-        """
-        return self._pag_scale > 0 and len(self.pag_applied_layers) > 0
-
-    @property
-    def pag_attn_processors(self):
+    def set_pag_applied_layers(
+        self,
+        pag_applied_layers: Union[str, List[str]],
+        pag_attn_processors: Tuple[AttentionProcessor, AttentionProcessor] = (
+            PAGCFGIdentitySelfAttnProcessor2_0(),
+            PAGIdentitySelfAttnProcessor2_0(),
```

Comment:
There should Hunyuan processors as well, right?

Reply:
Yes, correct. Will update

Reply:
Actually, no. If this was a type hint, we would have had Hunyuan ones as well. I am instantiating with default pag attention processors here if the user does not provide any (which is how we do it in most pag pipelines). For Hunyuan, we explicitly pass the pag_attn_processors parameter.